libs/pageserver_api/src/config.rs

```diff
@@ -296,7 +296,13 @@ pub mod defaults {
 
     pub const DEFAULT_INGEST_BATCH_SIZE: u64 = 100;
 
-    pub const DEFAULT_MAX_VECTORED_READ_BYTES: usize = 128 * 1024; // 128 KiB
+    // Soft limit for the maximum size of a vectored read.
+    // This is determined by the largest NeonWalRecord that can exist (minus dbdir and reldir keys
+    // which are unbounded). As of this writing, that is a `NeonWalRecord::ClogSetCommitted` record,
```

Comment:
at the blob storage level we have a limit of 256 MB since https://github.com/neondatabase/neon/pull/7852 (warning logged) and https://github.com/neondatabase/neon/pull/8106 (error returned).

```suggestion
    // which are bounded by the blob io limits only). As of this writing, that is a `NeonWalRecord::ClogSetCommitted` record,
```

Reply:
https://github.com/neondatabase/neon/pull/9221/commits/1464ba15e4cfe5ad39139a4e76818d024ac923d0